sailthru_content/services/catalog_api_service.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+from edx_rest_api_client.client import EdxRestApiClient
+
+
+logger = logging.getLogger()
+COURSES_PAGE_SIZE = 500
+PROGRAMS_PAGE_SIZE = 40
+
+
+class CatalogApiService(object):
+
+    _cached_programs = {}
```

Comment:
Mutable class attributes are dangerous! If someone modifies this attribute by accessing it through a particular instance, the referenced object accessed by all other instances of the class will be changed.

This should be an instance attribute set during initialization. Also, don't call something cached if it's not really cached.